checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -226,10 +252,43 @@ private void runInference() {
     addMemberAndClassAnnotations();
   }
 
+  /**
+   * Analyzes a return statement and performs two computations:
+   *
+   * <ul>
+   *   <li>If the return node is a field with non-empty must-call obligations, adds a {@link
+   *       NotOwning} annotation to the return type of the current method.
```

Comment:
Shouldn't this only occur if, in a given method, *every* returned expression has this property? Do we have a test for a method that returns a field in one place, and return something else (e.g., something with an obligation) at a different return site? We should add such a test if not.

Reply:
Based on the algorithm we don't need to check all return statements but if you think it might be a common pattern I can change this part.

Reply:
Sorry, I think I misremembered the algorithm here. I checked the paper, and you are correct. Let's add a test for this situation anyway (i.e., a getter that sometimes returns a field and sometimes returns something else, like `null`) and make sure `@NotOwning` is inferred (the code here should do it, but it's good to have the regression test).